File: src/test/java/utils/bytes/HammingTest.java
package utils.bytes;

import utils.codec.Ascii;
import org.junit.Test;
import static org.junit.Assert.*;

public class HammingTest {
    @Test
    public void hammingDistanceTest() {
        byte[] fst = Ascii.fromAscii("this is a test");
        byte[] snd = Ascii.fromAscii("wokka wokka!!!");

        assertEquals(37, Hamming.distance(fst, snd));
    }
}

File: src/test/kotlin/utils/bytes/HammingTest.kt
package utils.bytes

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test
import utils.bytes.Hamming.distance
import utils.codec.Ascii.fromAscii

@ExperimentalUnsignedTypes
class HammingTest {
    @Test
    fun hammingDistanceTest() {
        val fst = fromAscii("this is a test")
        val snd = fromAscii("wokka wokka!!!")
        distance(fst, snd) shouldBe 37
    }
}